lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -4423,56 +4437,22 @@ mkApiTransaction timeInterpreter wrk wid timeRefLens tx = do
             $ W.shelleyOnlyReadRewardAccount @s @k @n db wid
         pure $ mkApiAnyCertificate rewardAccount derivPath <$> certificates
 
-    depositIfAny :: Natural
-    depositIfAny
-        | tx ^. (#txMeta . #direction) == W.Outgoing =
-            if totalIn < totalOut
-            then 0
-            else totalIn - totalOut
-        | otherwise = 0
-
-    -- (pending) when reclaim is coming we have (fee+out) - inp = deposit
-    -- tx is incoming, and the wallet spent for fee and received deposit - fee as out
-    -- (inLedger) when reclaim is accommodated we have out - inp < deposit as fee was incurred
-    -- So in order to detect this we need to have
-    -- 1. deposit
-    -- 2. have inpsWithoutFee of the wallet non-empty
-    -- 3. outs of the wallet non-empty
-    -- 4. tx Incoming
-    -- 5. outs - inpsWithoutFee <= deposit
-    -- assumption: this should work when
-    depositValue :: Natural
-    depositValue = fromIntegral . unCoin $ tx ^. #txDeposit
-
-    reclaimIfAny :: Natural
-    reclaimIfAny
-        | tx ^. (#txMeta . #direction) == W.Incoming =
-              if ( totalIn > 0 && totalOut > 0 && totalOut > totalIn)
-                 && (totalOut - totalIn <= depositValue) then
-                  depositValue
-              else
-                  0
-        | otherwise = 0
-
-    totalIn :: Natural
-    totalIn
-        = sum (txOutValue <$> mapMaybe snd (tx ^. #txInputs))
-        + sum (fromIntegral . unCoin <$> Map.elems (tx ^. #txWithdrawals))
-
-    totalOut :: Natural
-    totalOut
-        = sum (txOutValue <$> tx ^. #txOutputs)
-        + maybe 0 (fromIntegral . unCoin) (tx ^. #txFee)
-
-    txOutValue :: TxOut -> Natural
-    txOutValue = fromIntegral . unCoin . TxOut.coin
-
     toAddressAmountNoAssets
         :: TxOut
         -> AddressAmountNoAssets (ApiT Address, Proxy n)
     toAddressAmountNoAssets (TxOut addr (TokenBundle.TokenBundle coin _)) =
         AddressAmountNoAssets (ApiT addr, Proxy @n) (mkApiCoin coin)
 
+    depositEventTaken :: Tx.DepositEvent -> Quantity "lovelace" Natural
+    depositEventTaken = \case
+      Tx.DepositTaken coin -> Quantity (fromIntegral (unCoin coin))
+      Tx.DepositReturned _coin -> Quantity 0
```

Comment:
Hi @Unisay!

Suggestion: You might be able to use `Coin.toQuantity` here, instead of unwrapping and using `fromIntegral`. Otherwise, there is `Coin.toQuantityMaybe` (for conversions that may fail).

Also, please would it be possible to stick to 4 spaces for this indent? Thanks! 🙏🏻 

```suggestion
        Tx.DepositTaken coin -> Coin.toQuantity coin
        Tx.DepositReturned _coin -> Quantity 0
```